tests/test_pipeline.py

```diff
@@ -55,3 +91,19 @@ def test_image_to_text_vit(self):
         model = "nlpconnect/vit-gpt2-image-captioning"
         expected_result = "a soccer game with a player jumping to catch"
         self._test_image_to_text(model, expected_result)
+
+    def test_text_to_speech_speecht5(self):
+        model = "microsoft/speecht5_tts"
+        expected_result = 16000
+        self._test_text_to_speech(model, expected_result)
+
+    def test_text_to_speech_m4t(self):
+        model = "facebook/hf-seamless-m4t-medium"
+        expected_result = 16000
+        if packaging.version.parse(transformers.__version__) >= packaging.version.parse("4.40.0"):
+            self._test_text_to_speech(model, expected_result)
```

Comment:
`Traceback (most recent call last):
  File "/root/optimum-habana/tests/test_pipeline.py", line 104, in test_text_to_speech_m4t
    self._test_text_to_speech(model, expected_result)
  File "/root/optimum-habana/tests/test_pipeline.py", line 81, in _test_text_to_speech
    output = generator(text, forward_params=forward_params, generate_kwargs=generate_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/text_to_audio.py", line 182, in __call__
    return super().__call__(text_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1242, in __call__
    return self.run_single(inputs, preprocess_params, forward_params, postprocess_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1249, in run_single
    model_outputs = self.forward(model_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1149, in forward
    model_outputs = self._forward(model_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/text_to_audio.py", line 143, in _forward
    output = self.model.generate(**model_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/seamless_m4t/modeling_seamless_m4t.py", line 762, in gaudi_SeamlessM4TForTextToSpeech_generate
    text_generation_output = super(SeamlessM4TForTextToSpeech, self).generate(input_ids, **kwargs_text)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/generation/utils.py", line 805, in generate
    generation_mode = self._get_generation_mode(generation_config, assistant_model)
  File "/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/core/weight_sharing.py", line 73, in wrapped__getattr__
    result = self.original__get_attr__(name)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1691, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'SeamlessM4TForTextToSpeech' object has no attribute '_get_generation_mode'
`

I see the above issue when running the m4t test after downloading transformers 4.40.0. The other tests seem to pass

Reply:
It is caused by https://github.com/huggingface/transformers/pull/29441/files. and should be fixed when upgrading transformers in optimum-habana

Reply:
Let's wait for the upgrade to Transformers v4.40 in Optimum Habana before merging this one then